addons_engine/addons-manager/services/addons_runner.py

```diff
@@ -0,0 +1,291 @@
+import logging
+import os
+import threading
+
+import docker
+import requests
+from db import addons_db
+
+addons_manager = None
+
+DEFAULT_PROJECT_NAME = os.environ.get("DEFAULT_PROJECT_NAME") or "root_orchestrator"
+DEFAULT_NETWORK_NAME = f"{DEFAULT_PROJECT_NAME}_default"
+
+ADDON_ID_LABEL = os.environ.get("ADDON_ID_LABEL") or "oak.addon.id"
+ADDON_MANAGER_LABEL = os.environ.get("ADDON_MANAGER_KEY") or "oak.plugin.manager.id"
+ADDON_SERVICE_NAME_LABEL = os.environ.get("ADDON_SERVICE_NAME_LABEL") or "oak.service.name"
+
+MARKETPLACE_API = f"{os.environ.get('MARKETPLACE_ADDR')}/api/v1/marketplace"
+
+
+class DockerAddonRunner:
+    def __init__(self, manager_id):
+        self._client = docker.from_env()
+        self._manager_id = manager_id
+
+    def _get_networks(self):
+        return [network.name for network in self._client.networks.list()]
+
+    def _create_network(self, network_name, **kwargs):
+        self._client.networks.create(network_name, **kwargs)
+
+    def _get_container(self, container_name):
+        try:
+            return self._client.containers.get(container_name)
+        except Exception:
+            return None
+
+    def _stop_container(self, container):
+        container.stop()
+
+    def _remove_container(self, container):
+        container.remove()
+
+    def _get_container_networks(self, container):
+        networks = container.attrs["NetworkSettings"]["Networks"]
+        return list(networks.keys())
+
+    def _get_container_ports(self, container):
+        return container.attrs["NetworkSettings"]["Ports"]
+
+    def maybe_create_networks(self, networks):
```

Comment:
nitpick:

"maybe_create_networks" sounds strange to me. How about "fix_unavailable_networks" or "try_to_fix_unavailable_networks"